block.go

```diff
@@ -0,0 +1,99 @@
+package bc
+
+import (
+	"encoding/hex"
+
+	"github.com/libsv/go-bt"
+)
+
+/*
+Field 													Purpose 									 														Size (Bytes)
+----------------------------------------------------------------------------------------------------
+block_header 					Block Header				 																									80
+txn_count 						Total number of txs in this block, including the coinbase tx 	 				VarInt
+txns 									Every tx in this block, one after another, in raw tx format 	 				-
+*/
+
+// A Block in the Bitcoin blockchain.
+type Block struct {
+	BlockHeader *BlockHeader
+	Txs         []*bt.Tx
+}
+
+// String returns the Block Header encoded as hex string.
+func (b *Block) String() (string, error) {
+	bb, err := b.Bytes()
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bb), nil
+}
+
+// Bytes will decode a bitcoin block struct into a byte slice.
+//
+// See https://btcinformation.org/en/developer-reference#serialized-blocks
+func (b *Block) Bytes() ([]byte, error) {
+	bytes := []byte{}
+
+	bh, err := b.BlockHeader.Bytes()
+	if err != nil {
+		return nil, err
+	}
+	bytes = append(bytes, bh...)
+
+	txCount := uint64(len(b.Txs))
+	bytes = append(bytes, bt.VarInt(txCount)...)
+
+	for _, tx := range b.Txs {
+		bytes = append(bytes, tx.ToBytes()...)
+	}
+
+	return bytes, nil
+}
+
+// NewBlockFromStr will encode a block header hash
+// into the bitcoin block header structure.
+//
+// See https://btcinformation.org/en/developer-reference#serialized-blocks
+func NewBlockFromStr(blockStr string) (*Block, error) {
+	blockBytes, err := hex.DecodeString(blockStr)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewBlockFromBytes(blockBytes)
+}
+
+// NewBlockFromBytes will encode a block header byte slice
+// into the bitcoin block header structure.
+//
+// See https://btcinformation.org/en/developer-reference#serialized-blocks
+func NewBlockFromBytes(b []byte) (*Block, error) {
+
+	var offset int
+	bh, err := NewBlockHeaderFromBytes(b[:80])
+	if err != nil {
+		return nil, err
+	}
+	offset += 80
+
+	txCount, size := bt.DecodeVarInt(b[offset:])
+	offset += size
+
+	var txs []*bt.Tx
+
```

Comment:
There's a lot of whitespace here that could be removed, makes the function about 7 lines longer than it needs to be.

Reply:
i like whitespace to separate different entities - for example blockheader, whitespace, txcount, whitespace, txs, etc. think it makes the code easier to read than just lumping up everything together, you don't think so? (don't really see anything wrong with if the code is a bit longer than it can be)

Reply:
It's just a preference thing, I don't really like a lot of whitespace in code unless it's separate blocks like if statements or looping.